test/integration/collada_world_exporter.cc

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <gtest/gtest.h>
+
+#include <ignition/common/Console.hh>
+#include <ignition/common/Filesystem.hh>
+
+#include "ignition/gazebo/Server.hh"
+#include "ignition/gazebo/test_config.hh"
+
+#include "helpers/UniqueTestDirectoryEnv.hh"
+
+using namespace ignition;
+using namespace gazebo;
+
+/////////////////////////////////////////////////
+class ColladaWorldExporterFixture : public ::testing::Test
+{
+  public: void SetUp() override
+  {
+    ignition::common::Console::SetVerbosity(4);
+  }
+
+  public: void LoadWorld(const std::string &_path)
+  {
+    ServerConfig serverConfig;
+    serverConfig.SetResourceCache(test::UniqueTestDirectoryEnv::Path());
+    serverConfig.SetSdfFile(common::joinPaths(PROJECT_SOURCE_PATH, _path));
+
+    std::cout << "Loading: " << serverConfig.SdfFile() << std::endl;
+    this->server = std::make_unique<Server>(serverConfig);
+    EXPECT_FALSE(server->Running());
+  }
+
+  public: std::unique_ptr<Server> server;
+};
+
+/////////////////////////////////////////////////
+TEST_F(ColladaWorldExporterFixture, ExportWorld)
+{
+  this->LoadWorld("test/worlds/collada_world_exporter.sdf");
+
+  // The export directory shouldn't exist.
+  EXPECT_FALSE(common::exists("./collada_world_exporter_box_test"));
+
+  // Run one iteration which should export the world.
+  server->Run(true, 1, false);
+
+  // The export directory should now exist.
+  EXPECT_TRUE(common::exists("./collada_world_exporter_box_test"));
+
+  // Cleanup
+  common::removeAll("./collada_world_exporter_box_test");
```

Comment:
The test is currently saving the file to the directory where the test is ran from. If there's any problem and the test starts failing, there's a chance these files will be left in the filesystem. I'd recommend at least saving to `PROJECT_BINARY_PATH` like the log test does:

https://github.com/ignitionrobotics/ign-gazebo/blob/8fd9b2647a5c37b1dde5509b5a741c6c955ee10f/test/integration/log_system.cc#L263-L264

In the future, we may use a strategy like the one introduced in https://github.com/ignitionrobotics/ign-common/pull/158.

Reply:
I don't think I get set where the test exports the mesh.

Reply:
I added a cleanup call before the exporter starts.

Reply:
Maybe the working directory could be changed with [common::chdir`](https://github.com/ignitionrobotics/ign-common/blob/a8e59e10b2ad80e7e52523b4869402acbfae5650/include/ignition/common/Filesystem.hh#L159-L162)